src/propagation.cpp

```diff
@@ -272,6 +292,12 @@ ot::expected<void> SpanContext::serialize(const ot::TextMapWriter &writer,
       if (!result) {
         return result;
       }
+      if (!origin_.empty()) {
+        result = writer.Set(headers_impl.origin_header, origin_);
+        if (!result) {
+          return result;
+        }
+      }
     } else if (nginx_opentracing_compatibility_hack_) {
       // See the comment in the header file on nginx_opentracing_compatibility_hack_.
       result = writer.Set(headers_impl.sampling_priority_header, "1");
```

Comment:
Can we add an nginx integration test? Then we'll know if we need to add it here or not.

The issue is that nginx needs to have a whitelist of any headers used, and it gets this via the hax of creating a dummy span and looking at what headers are on that span. Since it's dummy, it's not going to receive the "origin" header, so then it's not going to propagate it either. So then when you DO get one with an origin header, then it'll fail. Hence this section of the code.

Reply:
Updated the nginx_integration_test script and added tests for origin header and tags.